### DISCUSSION TITLE
Command to toggle Book View start point

### DISCUSSION THREAD

**Discussion by AKimura** [sumatrapdfreader/sumatrapdf#3220] [Ideas]
I'm a big fan of SumatraPDF and having a request.
SumatraPDF actually has two Book View variations, one of them is just the "Book View", other one is "Facing".

I eager to have a shortcut to toggle them.
There're shortcuts Ctrl+7 to switch to Facing, and Ctrl + 8 to switch to Book View already, but seems to be  no option to toggle them.
Please think a situation below:

When somebody is reading a pdf book, he won't being aware of which view mode is active.
Suddenly he notices the page's division point is not confortable, getting to want to TOGGLE the book's left page is odd or even numbered page.

Isn't this emotion common?
I know there's alerady a command "CmdToggleMangaMode", but that is only to switch left and right page.
This is not useful for the situations like above.
What we want to change is not the book's paging direction but the start point of the double-page spread.

Please consider my request.
I bilieve this is kind of universal demand and makes your app even better (though it's special already).

**Comment by GitHubRulesOK** [comment]
If the problem is spread is preferred, then you can set your default opening layout to book view in Options but if an author has set viewing spreads to facing then you only need to ctrl+7 as required. To add your own toggle you could consider autohotkey as a means to combine 2 alternative key actions

**Reply by AKimura** [reply]
To elaborate, the situation I had in mind was to read PDF scores practicing instruments.
I shouold've said this first, maybe.

You may want to practice the section A from P.1 to P.2, and then want to practice some solo section from P.4 to P.5, and then some section from P.7 to P.8...
I think you'd agree in this situation you exactly need the toggle function.

For the time being I'll test autohotkey.
Thanks!

**Reply by GitHubRulesOK** [reply]
hmm your usage is natural for that scenario of music sheets and 2 separated key choice is more difficult than 1 key pair but you can reassign those commands to single keys of your choose from those not already used say v and b are easy  if View > Book mean something different in your mind so touch one or other is easy if the page needs flipping between them
```
Shortcuts [
	[
		Cmd = CmdFacingView
		Key = v
	]
	[
		Cmd = CmdBookView
		Key = b
	]
]
```

**Reply by AKimura** [reply]
Thank you so much.
One more thing, very sorry for giving out information bit by bit, but to tell you the truth I'm using a usb triple foot pedal like this.
https://www.amazon.com/-/es/RI-FP3BK/dp/B004HAFZ1I

After applyng page down/up function to two pedals, I've got noticed what should be the last one pedal's function.
So, this may a little far from "universal demand", but at least just for me it must be a toggle function...

But thanks to you I could start using AutoHotKey (ver 2) and make a tiny script for this purpose.

```
^t::
{
	callCount := RegRead("HKEY_CURRENT_USER\SOFTWARE\SumatraPDF Script", "callCount", 0)
	callCount++
	RegWrite callCount, "REG_DWORD", "HKEY_CURRENT_USER\SOFTWARE\SumatraPDF Script", "callCount"
	
	if Mod(callCount, 2) == 1
	{
		SendInput "^{7}"
	}
	else
	{
		SendInput "^{8}"
	}
}
```

Using registry feels not good but this is my limit for now.
I'll remain this code for similar people ;)

**Reply by GitHubRulesOK** [reply]
hmm brilliant concept so laterally thinking I could have suggested use external viewer to get current page and add one via cmd HOWEVER that would require some error checks to not add last page+1 😃 nor perhaps be as easy since those shortcuts need 3 keys 

There is perhaps a means to read current page in AHK from toolbar if memory serves me right when using old nod5 helpers (there were 2 but see this line `ControlGetText, xedit, Edit1, A 	;page num from editbox (works even when edit is hidden)` https://github.com/nod5/sumatra_highlight_helper/blob/db6587121c03d23852ba00cd915f76574453a8d0/sumatra_highlight_helper.ahk#L95

hmm just looking flip page sides does NOT change page number in 2 page case, but then again in more than two pages does show if odd or even so can be used to externally re-call `-view facing` or `-view "book view"` especially if combined with correct page adjusted number.

```
"single page"
"continuous single page"
facing
"continuous facing"
"book view"
"continuous book view"
```

**Reply by AKimura** [reply]
Thank you for new suggestions.
Seems there's room for improvement as I expected.
But I'm a little frightened this would be unmanageable for me.

For now I'll endure with my AHK script and look forward to future SumatraPDF's official inplimentation.
Thanks again!
